test-bash.sh

```diff
@@ -44,6 +44,70 @@ test_resolve_symlinks() {
     [ "$(resolve_symlinks /tmp/1/2/foo)" == "/tmp/foo" ]
 }
 
+test_join_standard() {
+    local result
+
+    result=$(join "def" - 1 2 3)
+
+    [ "$result" == "1-2-3" ]
+}
+
+test_join_one() {
+    local result
+
+    result=$(join "def" - 1)
+
+    [ "$result" == "1" ]
+}
+
+test_join_spaces() {
+    local result
+
+    result=$(join "def" - 1 '' '' 3)
+
+    [ "$result" == "1-3" ]
+}
+
+test_join_more_spaces() {
+    local result
+
+    result=$(join "def" - 1   3)
+
+    [ "$result" == "1-3" ]
+}
+
+test_join_other_delimeter() {
+    local result
+
+    result=$(join "def" "b" 1   3)
+
+    [ "$result" == "1b3" ]
+}
+
+test_join_invalid_delimeter() {
+    local result
+
+    result=$(join "def" "bb" 1   3)
+
+    [ "$?" == 1 ]
+}
+
+test_join_no_delimeter() {
+    local result
+
+    result=$(join "def")
+
+    [ "$result" == "def" ]
+}
+
+test_join_default() {
+    local result
+
+    result=$(join "def" - '' '')
+
+    [ "$result" == "def" ]
+}
+
```

Comment:
Can you add the following test for joining an element containing a newline?

```bash
test_join_multiline_string_element() {
    local result
    local -a test_input=(
        "foo"
        "bar"
        "omg"$'\n'"baz"
    )

    result=$(join "def" "," "${test_input[@]}")

    [ "$result" == "foo,bar,omg"$'\n'"baz" ]
}
```

The current code fails this test.